lib/resque/worker.rb

```diff
@@ -651,7 +651,7 @@ def unregister_worker(exception = nil)
       message = exception_while_unregistering.message
       if exception
         message = message + "\nOriginal Exception (#{exception.class}): #{exception.message}\n" +
-                            "  #{exception.backtrace.join("  \n")}"
+                            "  #{(exception.backtrace || []).join("  \n")}"
```

Comment:
There is no need for "\n  " on the end in this case.  I would instead this and the above line to

```ruby
message += "\nOriginal Exception (#{exception.class}): #{exception.message}"
message += "\n  #{exception.backtrace.join("  \n")}" if exception.backtrace
```